packages/components/src/segmented-control/types.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { MutableRefObject, Ref, ReactNode, ReactText } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import type { RadioStateReturn } from 'reakit';
+
+/**
+ * Internal dependencies
+ */
+import type { FormElementProps } from '../utils/types';
+
+/**
+ * Option to render within `SegmentedControl`.
+ *
+ * @example
+ * ```jsx
+ * const option = { value: 'elsa', label: 'Elsa' };
+ * ```
+ */
+export type SegmentedControlOption = {
```

Comment:
I guess this should be
```suggestion
export type SegmentedControlOptionProps = {
```